cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -590,5 +592,58 @@ struct duration_scalar : chrono_scalar<T> {
    */
   typename T::rep count() { return this->value().count(); }
 };
+
+/**
+ * @brief An owning class to represent a list value in device memory
+ */
+class list_scalar : public scalar {
+ public:
+  using value_type = cudf::list_view;
+
+  list_scalar() : scalar(data_type(type_id::LIST)) {}
+  ~list_scalar()                        = default;
+  list_scalar(list_scalar&& other)      = default;
+  list_scalar(list_scalar const& other) = default;
+  list_scalar& operator=(list_scalar const& other) = delete;
+  list_scalar& operator=(list_scalar&& other) = delete;
+
+  /**
+   * @brief Construct a new list scalar object from existing device data
+   *
+   * @param value The value of the list
+   * @param is_valid Whether the value held by the scalar is valid
+   * @param stream CUDA stream used for device memory operations.
+   * @param mr Device memory resource to use for device memory allocation
+   */
+  list_scalar(cudf::column_view const& value,
+              scalar_validity validity            = scalar_validity::VALID,
```

Comment:
Do we really need a new validity struct? I assume we must have one already that is general enough to work in this use case as well.

Reply:
Searching through `types.hpp` and don't seem to have an enum that represents a VALID/INVALID state. Previous scalar ctors uses a boolean here. We can follow, just not sure if this is an exception to the taboo of using Boolean as argument type.